### ISSUE TITLE
Problem with optimization: fmin_l_bfgs_b terminated abnormally...

### ISSUE THREAD

**Issue by Ricardus312** [fmfn/BayesianOptimization#74]
I'm running the BO to search for hyperparameters of XGBoost model.  On the first iteration of the BO, after initialization, I get this message:

/home/ubuntu/anaconda3/lib/python3.6/site-packages/sklearn/gaussian_process/gpr.py:457: UserWarning: fmin_l_bfgs_b terminated abnormally with the  state: {'grad': array([ -1.36050603e-05]), 'task': b'ABNORMAL_TERMINATION_IN_LNSRCH', 'funcalls': 51, 'nit': 4, 'warnflag': 2}
  " state: %s" % convergence_dict)

More information about this issue:
https://github.com/scikit-learn/scikit-learn/issues/7563
https://stackoverflow.com/questions/34663539/scipy-optimize-fmin-l-bfgs-b-returns-abnormal-termination-in-lnsrch

I was using the default gp_params. If I understood correctly those two threads, the problem is that lbfgs received a negative gradient while trying to maximize the function, which might be caused because "the gradient does not match the function" or a wrong set of parameters for the Gaussian Process. The thing is that I have no idea of how to solve this, so any help is appreciated.

**Comment by fmfn** [comment]
As you pointed out yourself, this is a problem with training the guassian process, not with BO per se. As suggested, the likely cause is the function being optimized (the one the GP is trying to approximate) is ill behaved, leading to difficulties. You can try sharing some code, but the answer seems to be tuning GP's parameters until you get it working. Probably adding flexibility with the `alpha` parameter will help.

**Comment by Ricardus312** [comment]
Raising the value of alpha to 1e-4 solved the problem. Thank you very much for the help.

**Comment by fmfn** [comment]
@iliatimofeev This example is a bit outdated, it was created before the new scikit-learn implementation of gaussian processes. See @Ricardus312 comment above for a solution.

Examples will be reworked/fixed in an upcoming version release.
